src/util.jl

```diff
@@ -25,6 +25,7 @@ Return a CartesianIndex of dimension `N` which is one at index `i` and zero else
 Return CartesianIndices range excluding a single layer of cells on all boundaries.
 """
 @inline inside(a::AbstractArray;buff=1) = CartesianIndices(map(ax->first(ax)+buff:last(ax)-buff,axes(a)))
+@inline inside(dims::NTuple{N}) where N = CartesianIndices(map(i->(2:i-1),dims))
```

Comment:
Why do we need to add this?

Reply:
I added this line originally for the situation when gravity has the component in the periodic BC.
But yeah, add the value to all component including ghost cells does not matter.